simulator/mips/t/unit_test.cpp

```diff
@@ -243,3 +206,454 @@ TEST_CASE( "MIPS32_instr_disasm: Process_Disasm_Branches")
+    CHECK( instr.get_v_dst() == 0x100);
+
+    // Little endian
+    instr.set_v_src( 0xff000000, 0);
+    instr.set_v_src( 4, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0xfff00000);
+}
+
+TEST_CASE( "MIPS32_instr: jr")
+{
+    MIPS32Instr instr( "jr");
+
+    instr.set_new_PC( instr.get_PC() + 4);
+    instr.set_v_src( 1024, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == 1024);
+
+    instr.set_new_PC( instr.get_PC() + 4);
+    instr.set_v_src( instr.get_PC() + 2, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == instr.get_PC() + 4);
+}
+
+TEST_CASE( "MIPS32_instr: jalr")
+{
+    MIPS32Instr instr( "jalr");
+
+    instr.set_new_PC( instr.get_PC() + 4);
+    instr.set_v_src( instr.get_PC() + 8, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == instr.get_PC() + 8);
+    CHECK( instr.get_v_dst() == instr.get_PC() + 4);
+
+    instr.set_new_PC( instr.get_PC() + 4);
+    instr.set_v_src( instr.get_PC() + 2, 0);
+    instr.execute();
+    CHECK( instr.get_new_PC() == instr.get_PC() + 4);
+}
+
+/*
+TEST_CASE( "MIPS32_instr: movz")
+{
+    MIPS32Instr instr( "movz");
+
+    instr.set_v_src( 3, 0);
+    instr.set_v_src( 2, 1);
+    instr.set_v_dst( 4);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 4);    // fails ( 3 == 4)
+
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 1, 1);
+    instr.set_v_dst( 4);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 4);    // fails ( 1 == 4)
+    
+    instr.set_v_src( 0, 0);
+    instr.set_v_src( 0, 1);
+    instr.set_v_dst( 4);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0);
+
+    instr.set_v_src( 3, 0);
+    instr.set_v_src( 0, 1);
+    instr.set_v_dst( 4);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 3);
+}
+
+TEST_CASE( "MIPS32_instr: movn")
+{
+    MIPS32Instr instr( "movn");
+
+    instr.set_v_src( 3, 0);
+    instr.set_v_src( 0, 1);
+    instr.set_v_dst( 4);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 4);    // fails ( 3 == 4)
+
+    instr.set_v_src( 3, 0);
+    instr.set_v_src( 2, 1);
+    instr.set_v_dst( 4);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 3);
+}
+*/
+
+// currently disabled in TT
+TEST_CASE( "MIPS32_instr: syscall")
+{
+    MIPS32Instr instr( "syscall");
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::SYSCALL);
+}
+
+TEST_CASE( "MIPS32_instr: break")
+{
+    MIPS32Instr instr( "break");
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::BREAKPOINT);
+}
+
+// Not implemented in SPIM TT
+TEST_CASE( "MIPS32_instr: mfhi")
+{
+    MIPS32Instr instr( "mfhi");
+    instr.set_v_dst( 0);
+    instr.set_v_src( 0xfee1dead, 0);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0xfee1dead);
+}
+
+// Not implemented in SPIM TT
+TEST_CASE( "MIPS32_instr: mthi")
+{
+    MIPS32Instr instr( "mthi");
+    instr.set_v_dst( 0);
+    instr.set_v_src( 0xfee1dead, 0);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0xfee1dead);
+}
+
+// Not implemented in SPIM TT
```

Comment:
What do you mean by that comment?

Reply:
There is no test for this instruction in SPIM TT.

Reply:
OK. Probably that's not so important information to post.

Reply:
My point is that comment should draw attention to some danger or strangeness (you discussed it with Kirill). For that — it is not needed.